tools-local/tasks/mobile.tasks/WasmAppBuilder/WasmAssemblySearcher.cs

```diff
@@ -0,0 +1,159 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Reflection;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+public class WasmAssemblySearcher : Task
+{
+    [Required]
+    public string? MainAssembly { get; set; }
+
+    // If true, continue when a referenced assembly cannot be found.
+    // If false, throw an exception.
+    public bool SkipMissingAssemblies { get; set; }
+
+    // Either one of these two need to be set
+    public ITaskItem[]? AssemblySearchPaths { get; set; }
+    public ITaskItem[]? Assemblies { get; set; }
+    public ITaskItem[]? ExtraAssemblies { get; set; }
+    public int DebugLevel { get; set; }
+
+    // The set of assemblies the app will use
+    [Output]
+    public ITaskItem[]? ReferencedAssemblies { get; set; }
+
+    private SortedDictionary<string, Assembly>? _assemblies;
+    private Resolver? _resolver;
+
+    public override bool Execute ()
+    {
+        if (AssemblySearchPaths == null && Assemblies == null)
+            throw new ArgumentException("Either the AssemblySearchPaths or the Assemblies property needs to be set.");
+
+        var paths = new List<string>();
+        _assemblies = new SortedDictionary<string, Assembly>();
+
+        if (AssemblySearchPaths != null)
+        {
+            // Collect and load assemblies used by the app
+            foreach (var v in AssemblySearchPaths!)
+            {
+                var dir = v.ItemSpec;
+                if (!Directory.Exists(dir))
+                    throw new ArgumentException($"Directory '{dir}' doesn't exist or not a directory.");
+                paths.Add(dir);
```

Comment:
Once you change `AssemblySearchPaths` to `string[]`, you can drop `paths`.

Reply:
Thanks